src/accel/FastSimulationOffload.hh

```diff
@@ -0,0 +1,60 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/FastSimulationOffload.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <G4VFastSimulationModel.hh>
+#include <G4Version.hh>
+
+namespace celeritas
+{
+class SharedParams;
+class LocalTransporter;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Offload tracks to Celeritas via G4VFastSimulationModel interface
+ *
+ * This class must be constructed locally on each worker thread/task, typically
+ * within the application's concrete implementation of
+ * `G4VUserDetectorConstruction::ConstructSDandField()`.
+ */
+class FastSimulationOffload final : public G4VFastSimulationModel
+{
+  public:
+    //! Construct with name, shared (across threads) params, and thread-local
+    //! transporter
```

Comment:
The convention for Celeritas documentation is to put Doxygen comments with *definitions* not *declarations*: so the class doxygen goes in the header file (where the class itself is declared); inline functions in the header get short `//!` remarks; and non-inline functions get a `/*! */` definition in the `.cc` file. To make class definitions more readable we usually put a one-line non-doxygen comment above the member function declarations.